client/rpcserver/rpcserver_test.go

```diff
@@ -525,11 +518,19 @@ func TestAuthMiddleware(t *testing.T) {
 
 func TestClientMap(t *testing.T) {
 	s, _, shutdown := newTServer(t, true, "", "")
-	conn := new(TConn)
+	resp := make(chan []byte, 1)
+	conn := &TConn{respReady: resp}
+	// msg.ID == 0 gets an error response, which can be discarded.
+	read, _ := json.Marshal(msgjson.Message{ID: 0})
+	fmt.Println(string(read))
+	conn.addRead(read)
 
-	s.mtx.(*tRWMutex).locked = make(chan struct{})
 	go s.websocketHandler(conn, "someip")
-	<-s.mtx.(*tRWMutex).locked
+
+	// When a response to our dummy message is received, the client should be in
+	// RPCServer's client map.
+	msg := <-resp
+	fmt.Println("Response to dummy message received from server:", string(msg))
```

Comment:
Pls remove the fmt.Print statements added in this commit.  I just included those for illustrative and debugging purposes.